app/scripts/controllers/profile.js

```diff
@@ -78,6 +133,7 @@ profileController.controller("profile", ['UploadImage', 'User', '$firebaseAuth',
       var picture = $('#uploadedPicture').attr('src');
       if (picture) {
         User.updateProfilePicture(picture);
+		$scope.profileChangeSucceeded = true;
```

Comment:
This might not always be the case. The `updateProfilePicture` doesn't return anything and it could fail inside that method. It fails at this [line](https://github.com/joshcronin/chata/blob/master/app/scripts/factories/user.js#L98). You could try and pass in an optional callback `User.updateUsername($scope.user.username, $scope.onProfilePictureConfirm.bind(this))`;. 
Then call `callback()` in [here](https://github.com/joshcronin/chata/blob/master/app/scripts/factories/user.js#L102) and set the correct message.